flaky server unit test

```
Server server accepting sessions that haven't completed the handshake 

  rejects new connection attempts if the accept queue is full

  /home/travis/gopath/src/github.com/lucas-clemente/quic-go/server_test.go:779

panic: 

Your test failed.

Ginkgo panics to prevent subsequent assertions from running.

Normally Ginkgo rescues this panic so you shouldn't see it.

But, if you make an assertion in a goroutine, Ginkgo can't capture the panic.

To circumvent this, you should call

	defer GinkgoRecover()

at the top of the goroutine that caused this panic.

goroutine 124 [running]:

github.com/onsi/ginkgo.Fail(0xc000668000, 0x1284, 0xc000431ae8, 0x1, 0x1)

	/home/travis/gopath/pkg/mod/github.com/onsi/ginkgo@v1.12.0/ginkgo_dsl.go:267 +0xc8

github.com/onsi/ginkgo/internal/testingtproxy.(*ginkgoTestingTProxy).Fatalf(0xc00000f1a0, 0xbf6e35, 0x2e, 0xc000651270, 0x5, 0x5)

	/home/travis/gopath/pkg/mod/github.com/onsi/ginkgo@v1.12.0/internal/testingtproxy/testing_t_proxy.go:45 +0xc1

github.com/golang/mock/gomock.(*Controller).Call.func1(0xc00014f290, 0xb90000, 0xc000097910, 0xbe7e5d, 0x16, 0xc0001b1140, 0x1, 0x1, 0x0, 0x0, ...)

	/home/travis/gopath/pkg/mod/github.com/golang/mock@v1.4.0/gomock/controller.go:201 +0x486

github.com/golang/mock/gomock.(*Controller).Call(0xc00014f290, 0xb90000, 0xc000097910, 0xbe7e5d, 0x16, 0xc0001b1140, 0x1, 0x1, 0x10000000043a1ee, 0x0, ...)

	/home/travis/gopath/pkg/mod/github.com/golang/mock@v1.4.0/gomock/controller.go:217 +0xb4

github.com/lucas-clemente/quic-go.(*MockPacketHandlerManager).GetStatelessResetToken(0xc000097910, 0xc000431ad0, 0x4, 0x4, 0x0, 0x0)

	/home/travis/gopath/src/github.com/lucas-clemente/quic-go/mock_packet_handler_manager_test.go:92 +0x111

github.com/lucas-clemente/quic-go.(*baseServer).createNewSession(0xc000495e40, 0xced0c0, 0xc0004de2a0, 0x0, 0x0, 0x0, 0xc000431ac0, 0xa, 0xa, 0xc000431ab8, ...)

	/home/travis/gopath/src/github.com/lucas-clemente/quic-go/server.go:434 +0x139

github.com/lucas-clemente/quic-go.(*baseServer).handleInitialImpl(0xc000495e40, 0xc000651130, 0xc000021780, 0x0, 0x0, 0x0, 0x18)

	/home/travis/gopath/src/github.com/lucas-clemente/quic-go/server.go:388 +0x426

github.com/lucas-clemente/quic-go.(*baseServer).handlePacketImpl(0xc000495e40, 0xc000651130, 0x2)

	/home/travis/gopath/src/github.com/lucas-clemente/quic-go/server.go:331 +0x2d3

github.com/lucas-clemente/quic-go.(*baseServer).run(0xc000495e40)

	/home/travis/gopath/src/github.com/lucas-clemente/quic-go/server.go:204 +0x160

created by github.com/lucas-clemente/quic-go.listen

	/home/travis/gopath/src/github.com/lucas-clemente/quic-go/server.go:187 +0x4fb

--- a/server_test.go
+++ b/server_test.go
@@ -806,19 +806,14 @@ var _ = Describe("Server", func() {
 				return sess
 			}
 
-			var wg sync.WaitGroup
-			wg.Add(protocol.MaxAcceptQueueSize)
+			phm.EXPECT().GetStatelessResetToken(gomock.Any()).Times(protocol.MaxAcceptQueueSize)
+			phm.EXPECT().Add(gomock.Any(), gomock.Any()).Return(true).Times(2 * protocol.MaxAcceptQueueSize)
 			for i := 0; i < protocol.MaxAcceptQueueSize; i++ {
-				go func() {
-					defer GinkgoRecover()
-					defer wg.Done()
-					phm.EXPECT().GetStatelessResetToken(gomock.Any())
-					phm.EXPECT().Add(gomock.Any(), gomock.Any()).Return(true).Times(2)
-					serv.handlePacket(getInitialWithRandomDestConnID())
-					Consistently(conn.dataWritten).ShouldNot(Receive())
-				}()
+				serv.handlePacket(getInitialWithRandomDestConnID())
 			}
-			wg.Wait()
+
+			Eventually(func() int32 { return atomic.LoadInt32(&serv.sessionQueueLen) }).Should(BeEquivalentTo(protocol.MaxAcceptQueueSize))
+			Consistently(conn.dataWritten).ShouldNot(Receive())
 
 			p := getInitialWithRandomDestConnID()
 			hdr := parseHeader(p.data)

